src/TraceEvent/TraceLog.cs

```diff
@@ -230,26 +230,37 @@ public static TraceLogEventSource CreateFromEventPipeSession(EventPipeSession se
         public class EventPipeRundownConfiguration
         {
             internal readonly DiagnosticsClient m_client;
+            internal readonly IEnumerable<EventPipeProvider> m_providers;
 
-            private EventPipeRundownConfiguration(DiagnosticsClient client) { m_client = client; }
+            private EventPipeRundownConfiguration(DiagnosticsClient client, IEnumerable<EventPipeProvider> providers)
+            {
+                m_client = client;
+                m_providers = providers;
+            }
 
             /// <summary>
             /// No rundown will be requested, thus it may be impossible to symbolicate events. This is OK, if you don't
             /// require method/module info the captured events.
             /// </summary>
             public static EventPipeRundownConfiguration None()
             {
-                return new EventPipeRundownConfiguration(null);
+                return new EventPipeRundownConfiguration(null, null);
             }
 
             /// <summary>
             /// If the rundown is enabled and a DiagnosticsClient is given, TraceLog.CreateFromEventPipeSession will
             /// create an additional short-lived diagnostics session to load all module/method information up to that
             /// point.
             /// </summary>
-            public static EventPipeRundownConfiguration Enable(DiagnosticsClient client)
+            public static EventPipeRundownConfiguration Enable(DiagnosticsClient client, IEnumerable<EventPipeProvider> providers = null)
```

Comment:
I think it would be nice to not change the signature of this method, but instead add a `AddProvider` method.  If the method doesn't get called then the default configuration is returned.  Any calls to `AddProvider` eliminate the default configuration.  This simplifies things for those that don't want to touch the rundown configuration.

Reply:
the parameter had a default value (`null`) so it didn't really change the API but no problem, I've made the change you've requested.

Reply:
Thank you @vaind.  One other small request - would you mind adding a method comment block for the new method and just describe that its usage overrides the default rundown configuration?

Reply:
done